model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -240,8 +240,10 @@ class ModelReplicationServer(
     override suspend fun PipelineContext<Unit, ApplicationCall>.postRepositoryObjectsGetAll(repository: String) {
         checkPermission(ModelServerPermissionSchema.repository(repository).objects.read)
         runWithRepository(repository) {
-            val keys = call.receiveStream().bufferedReader().use { reader ->
-                reader.lineSequence().toHashSet()
+            val keys = withContext(Dispatchers.IO) {
+                call.receiveStream().bufferedReader().use { reader ->
+                    reader.lineSequence().toHashSet()
+                }
             }
             val objects = withContext(Dispatchers.IO) { modelClient.store.getAll(keys) }.checkValuesNotNull {
```

Comment:
Any reason not to combine the two `withContext` blocks into one?

Reply:
Done. For now they can go into one.

(Later model client should not be called directly from here.)